phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -1500,7 +1528,7 @@ && isPartialUncoveredIndexMutation(indexMetaData, miniBatchOp))) {
             onDupCheckTime += (EnvironmentEdgeManager.currentTimeMillis() - start);
         }
 
-        if (context.hasConditionalTTL) {
+        if (context.hasConditionalTTL && isStrictTTLEnabled(miniBatchOp)) {
```

Comment:
Do we need this check as I see in the method call also we check if strictTTL is enabled upfront?